### PR TITLE
chore(docs): add recipe for datadog

### PR DESCRIPTION
**Pull request by danielbdias**
This PR adds more documentation for the Datadog integration

To check the documentation locally go to `./docs` dir and run `npm install && npm start` command.

## Changes

- Updated `Configuration > Connecting to Datastores > Datadog` page
- Added `Examples & Tutorials > Recipes > Running Tracetest with Datadog`

## Fixes

- https://github.com/kubeshop/tracetest/issues/1969

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [x] updated the docs
- [ ] added a test

### CODE REVIEW

File: examples/tracetest-datadog/tracetest/tracetest.config.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+---
+# postgres:
+#   host: tt-postgres
+#   user: postgres
+#   password: postgres
+#   port: 5432
+#   dbname: postgres
+#   params: sslmode=disable
+postgresConnString: "host=tt-postgres user=postgres password=postgres port=5432 sslmode=disable"
```
Comment (danielbdias) [inline examples/tracetest-datadog/tracetest/tracetest.config.yaml:9]: If we don't release the Viper version, only using `postgresConnString` will work for now.

File: docs/docs/configuration/connecting-to-data-stores/datadog.md [hunk 2]
```diff
@@ -8,9 +8,20 @@ Examples of configuring Tracetest with Datadog can be found in the [`examples` f
 
 ## Configuring OpenTelemetry Collector to Send Traces to both Datadog and Tracetest
 
-In your OpenTelemetry Collector config file, make sure to set the `exporter` to `otlp/tt`, with the `endpoint` pointing to your Tracetest instance on port `21321`. If you are running Tracetest with Docker, the endpoint might look like this `http://tracetest:21321`.
+In your OpenTelemetry Collector config file:
 
-Additionally, set another `exporter` to `datadog`, with the `endpoint` pointing to your Datadog account. Set the site to the Datadog API `datadoghq.com` and add your API key.
+- Set the `exporter` to `otlp/tt`
```
Comment (jfermi) [inline docs/docs/configuration/connecting-to-data-stores/datadog.md:13]:
```suggestion
- Set the `exporter` to `otlp/tt`.
```

File: docs/docs/configuration/connecting-to-data-stores/datadog.md [hunk 3]
```diff
@@ -8,9 +8,20 @@ Examples of configuring Tracetest with Datadog can be found in the [`examples` f
...
+- Set the `endpoint` to your Tracetest instance on port `21321`
```
Comment (jfermi) [inline docs/docs/configuration/connecting-to-data-stores/datadog.md:14]:
```suggestion
- Set the `endpoint` to your Tracetest instance on port `21321`.
```

File: docs/docs/configuration/connecting-to-data-stores/datadog.md [hunk 4]
```diff
@@ -8,9 +8,20 @@ Examples of configuring Tracetest with Datadog can be found in the [`examples` f
...
+
+:::tip
+If you are running Tracetest with Docker, the endpoint might look like this `http://tracetest:21321`
```
Comment (jfermi) [inline docs/docs/configuration/connecting-to-data-stores/datadog.md:17]:
```suggestion
If you are running Tracetest with Docker, the endpoint might look like this `http://tracetest:21321`.
```

File: docs/docs/configuration/connecting-to-data-stores/datadog.md [hunk 5]
```diff
@@ -8,9 +8,20 @@ Examples of configuring Tracetest with Datadog can be found in the [`examples` f
...
+:::
+
+Additionally, add another config:
+
+- Set the `exporter` to `datadog`
```
Comment (jfermi) [inline docs/docs/configuration/connecting-to-data-stores/datadog.md:22]:
```suggestion
- Set the `exporter` to `datadog`.
```

File: docs/docs/configuration/connecting-to-data-stores/datadog.md [hunk 6]
```diff
@@ -8,9 +8,20 @@ Examples of configuring Tracetest with Datadog can be found in the [`examples` f
...
+- Set the `endpoint` pointing to your Datadog account
```
Comment (jfermi) [inline docs/docs/configuration/connecting-to-data-stores/datadog.md:23]:
```suggestion
- Set the `endpoint` pointing to your Datadog account.
```

File: docs/docs/configuration/connecting-to-data-stores/datadog.md [hunk 7]
```diff
@@ -8,9 +8,20 @@ Examples of configuring Tracetest with Datadog can be found in the [`examples` f
...
+- Set the site to the Datadog API `datadoghq.com` and add your API key
```
Comment (jfermi) [inline docs/docs/configuration/connecting-to-data-stores/datadog.md:24]:
```suggestion
- Set the site to the Datadog API `datadoghq.com` and add your API key.
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-datadog.md [hunk 8]
```diff
@@ -0,0 +1,338 @@
+# Running Tracetest With Datadog
+
+:::note
+[Check out the source code on GitHub here.](https://github.com/kubeshop/tracetest/tree/main/examples/tracetest-datadog)
+:::
+
+[Tracetest](https://tracetest.io/) is a testing tool based on [OpenTelemetry](https://opentelemetry.io/) that allows you to test your distributed application. It allows you to use your telemetry data generated by the OpenTelemetry tools to check and assert if your application has the desired behavior defined by your test definitions.
+
+[Datadog](https://www.datadoghq.com/) is an observability solution for cloud-scale applications, providing solutions to monitor databases, servers, tools and services. It provides an integrated distributed tracing, logs, and metrics solution and supports OpenTelemetry standards.
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-datadog.md:9]:
```suggestion
[Datadog](https://www.datadoghq.com/) is an observability solution for cloud-scale applications, providing solutions to monitor databases, servers, tools and services. It provides integrated distributed tracing, logs and metrics solution and supports OpenTelemetry standards.
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-datadog.md [hunk 9]
```diff
@@ -0,0 +1,338 @@
...
+
+## OpenTelemetry Demo `v1.3.0` with Datadog, OpenTelemetry and Tracetest
+
+This is a simple sample app on how to configure the [OpenTelemetry Demo `v1.3.0`](https://github.com/open-telemetry/opentelemetry-demo) to use [Tracetest](https://tracetest.io/) for enhancing your E2E and integration tests with trace-based testing, and [Datadog](https://www.datadoghq.com/) as a trace data store.
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-datadog.md:13]:
```suggestion
This is a simple sample app on how to configure the [OpenTelemetry Demo `v1.3.0`](https://github.com/open-telemetry/opentelemetry-demo) to use [Tracetest](https://tracetest.io/) for enhancing your E2E and integration tests with trace-based testing and [Datadog](https://www.datadoghq.com/) as a trace data store.
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-datadog.md [hunk 10]
```diff
@@ -0,0 +1,338 @@
...
+
+## Prerequisites
+
+You will need [Docker](https://docs.docker.com/get-docker/) and [Docker Compose](https://docs.docker.com/compose/install/) installed on your machine to run this sample app! Additionally, you will need a Datadog account and an API. Sign up to Datadog on their [homepage](https://www.datadoghq.com/) by clicking on `Get Started Free`.
+
+## Project structure
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-datadog.md:19]:
```suggestion
## Project Structure
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-datadog.md [hunk 11]
```diff
@@ -0,0 +1,338 @@
...
+          endpoint: otel-collector:4317
+
+server:
+  telemetry:
+    dataStore: otlp
+    exporter: collector
+    applicationExporter: collector
+```
+
+**How to send traces to Tracetest and Lightstep?**
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-datadog.md:179]:
```suggestion
**How to Send Traces to Tracetest and Lightstep**
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-datadog.md [hunk 12]
```diff
@@ -0,0 +1,338 @@
...
+      exporters: [otlp/tt]
+    traces/dd:
+      receivers: [otlp]
+      processors: [batch]
+      exporters: [datadog]
+```
+
+**Important!** Take a closer look at the sampling configs in both the `collector.config.yaml` and `tracetest.config.yaml`. They both set sampling to 100%. This is crucial when running trace-based e2e and integration tests.
+
+## Run both the OpenTelemetry Demo app and Tracetest
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-datadog.md:227]:
```suggestion
## Run Both the OpenTelemetry Demo App and Tracetest
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-datadog.md [hunk 13]
```diff
@@ -0,0 +1,338 @@
...
+
+To start both the OpenTelemetry and Tracetest we will run this command:
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-datadog.md:229]:
```suggestion
To start both the OpenTelemetry and Tracetest, we will run this command:
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-datadog.md [hunk 14]
```diff
@@ -0,0 +1,338 @@
...
+attr:tracetest.span.duration  < 10ms
+```
+
+![](../img/datadog-recipe-failed-test.png)
+
+Increasing the duration to a more reasonable `500ms` will make the test pass.
+
+![](../img/datadog-recipe-successful-test.png)
+
+## Run Tracetest tests with the Tracetest CLI
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-datadog.md:255]:
```suggestion
## Run Tracetest Tests with the Tracetest CLI
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-datadog.md [hunk 15]
```diff
@@ -0,0 +1,338 @@
...
+
+First, [install the CLI](https://docs.tracetest.io/getting-started/installation#install-the-tracetest-cli).
+Then, configure the CLI:
+
+```bash
+tracetest configure --endpoint http://localhost:11633 --analytics
+```
+
+Once configure, you can run a test against the Tracetest instance via the terminal.
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-datadog.md:264]:
```suggestion
Once configured, you can run a test against the Tracetest instance via the terminal.
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-datadog.md [hunk 16]
```diff
@@ -0,0 +1,338 @@
...
+                ✔ #90aeab1e9db4617b
+                        ✔ attr:rpc.grpc.status_code = 0 (http://localhost:11633/test/JBYAfKJ4R/run/3/test?selectedAssertion=1)
+        ✔ span[tracetest.span.type="rpc" name="hipstershop.ProductCatalogService/ListProducts" rpc.system="grpc" rpc.method="ListProducts" rpc.service="hipstershop.ProductCatalogService"]
+                ✔ #44b836b092b4d708
+                        ✔ attr:rpc.grpc.status_code = 0 (http://localhost:11633/test/JBYAfKJ4R/run/3/test?selectedAssertion=2)
+```
+
+If you edit the duration as in the Web UI example above, the test will pass!
+
+## View trace spans over time in Datadog
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-datadog.md:324]:
```suggestion
## View Trace Spans Over Time in Datadog
```

**Review by jfermi — COMMENTED** [review]
Some editing suggestions.

**Review by adnanrahic — APPROVED** [review]
This is amazing @danielbdias!

Agree with @jfermi's suggestions. Once those are fixed we can :shipit: !

FYI - I think we can merge this PR before #1984 
I can later fix any merge issues in my branch.

**Pull request merged**